### PR TITLE
docs: migration docs v4 to v5

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

closes: #1781
closed: #2083 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: docs/migrations/v4-to-v5.md
Comment (crodriguezvega) [inline docs/migrations/v4-to-v5.md:17]: Do we still need this line here?

**Comment by crodriguezvega** [comment]
Still a bit WiP, but open to receive feedback if I am forgetting something or if I am adding things in the wrong section.

File: docs/migrations/v4-to-v5.md
Comment (charleenfei) [reply docs/migrations/v4-to-v5.md:17]: i agree it could be removed, it's not very applicable to this file

File: docs/migrations/v4-to-v5.md
Comment (charleenfei) [reply docs/migrations/v4-to-v5.md:60]: i think it could be good to give an example of how it was previously in addition to what it should look like now

File: docs/migrations/v4-to-v5.md
Comment (damiannolan) [inline docs/migrations/v4-to-v5.md:164]: Should we update this with the new interface definition from #2058?

**Review by damiannolan — APPROVED** [review]
Awesome job! Just left one comment regarding the new interface type for `MsgServiceRouter`

File: docs/migrations/v4-to-v5.md
Comment (colin-axner) [inline docs/migrations/v4-to-v5.md:4]: I think we can probably remove this section going forward. I'd assume it is just noise nowadays

File: docs/migrations/v4-to-v5.md
Comment (colin-axner) [inline docs/migrations/v4-to-v5.md:35]: Can we link to the changes in the pr that changed this to show how it changed in our simapp app.go?

File: docs/migrations/v4-to-v5.md
Comment (colin-axner) [inline docs/migrations/v4-to-v5.md:10]: Food for thought, going forward we could add directories for each migration v4-to-v5 and within each directory is a file for each user. So the relayer only needs to open the relayer file and realize no changes should be necessary with respect to ibc-go

**Review by colin-axner — APPROVED** [review]
Excellent! This is by far the best migrations written thus far!

**Review by chatton — APPROVED** [review]
Really thorough great work @crodriguezvega 🥇

File: docs/migrations/v4-to-v5.md
Comment (crodriguezvega) [reply docs/migrations/v4-to-v5.md:35]: This actually changed twice, since we renamed the ante decorator twice... So I link to the [PR](https://github.com/cosmos/ibc-go/pull/1820) that did the last renaming?

Comment (colin-axner) [reply docs/migrations/v4-to-v5.md:35]: yes, sounds good to me! You can maybe comment on that section of the code in the pr that the name was changed twice

File: docs/migrations/v4-to-v5.md
Comment (colin-axner) [reply docs/migrations/v4-to-v5.md:10]: It might be nice on the [docs website](https://ibc.cosmos.network/main/migrations/v3-to-v4.html), if there was subsections that could be clicked on for each interested party

Comment (crodriguezvega) [reply docs/migrations/v4-to-v5.md]: I like @chatton's idea of having these as links. I just pushed an update.
We can iterate later on if we should move each to a separate file.

**Review by seantking — APPROVED** [review]

**Comment by colin-axner** [comment]
@crodriguezvega is this waiting on a review from someone?

File: docs/migrations/v4-to-v5.md
Comment (crodriguezvega) [reply docs/migrations/v4-to-v5.md]: Updated.

**Comment by crodriguezvega** [comment]
> @crodriguezvega is this waiting on a review from someone?

No, I just need to update according to @damiannolan's and yours comment.

File: docs/migrations/v4-to-v5.md
Comment (crodriguezvega) [reply docs/migrations/v4-to-v5.md:35]: Just realized that no changes were required actually in `app.go`, so I just added a link to the PR.

**Review by colin-axner — APPROVED** [review]

**Comment by crodriguezvega** [comment]
> Can we merge this?

Yeap. I will merge it now.

**Pull request merged**
